### PR TITLE
Food processor now transfers reagents and can be filled with a plant bag

### PR DESCRIPTION
**Pull request by Krimsonkhan**
I once made potent potatoes and was disappointed to discover that processing them into space fries resulted in less nutriment than eating them raw, so I made this. These changes are tested. 

:cl:
 * rscadd: Food processor now transfers reagents
 * rscadd Food processor can now be filled with a plant bag

### CODE REVIEW

**Comment by Kurfursten** [comment]
How does this interact with purifying food like removing poison from carp and spiders?

**Comment by Krimsonkhan** [comment]
> 
> 
> How does this interact with purifying food like removing poison from carp and spiders?

The purification code is still in place. This only transfers reagents for potatoes, carrots, soybeans, and meat (the meat that is processed into faggots) when processed.

**Comment by DamianX** [comment]
I quickly glanced at the code and it looks like you're making several food items lose all reagents except for 1u nutriment, is that right?
Won't that interfere with creating those items through admin magic or xenobiology or other ways?

**Comment by Krimsonkhan** [comment]
> 
> 
> I quickly glanced at the code and it looks like you're making several food items lose all reagents except for 1u nutriment, is that right?
> Won't that interfere with creating those items through admin magic or xenobiology or other ways?

It would, yes. Do you have any solutions in mind? I could add a flag that doesn't add the nutriments if the item was created via the food processor. Do you see any loopholes with that?

**Comment by Krimsonkhan** [comment]
Or just a processed variant that doesn't start have base nutriments; would probably be better than adding a spaghetti variable. I'll push these changes in a sec

**Review by DamianX — APPROVED** [review]
You could use `reagents.clear_reagents()` instead of removing each reagent manually. Other than that, this looks good.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -841,6 +841,11 @@
 	reagents.add_reagent(NUTRIMENT, 3)
 	bitesize = 2
 
+/obj/item/weapon/reagent_containers/food/snacks/faggot/processed/New()
+	..()
+	reagents.clear_reagents()
+	bitesize = 2
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Rereading this, I noticed there's really no reason to set this again, right? It's already set to 2 on the objects themselves?

Comment (Krimsonkhan) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: Will fix later today

**Pull request merged**